Lib/json/tool.py

```diff
@@ -31,21 +31,26 @@ def main():
     options = parser.parse_args()
 
     infile = options.infile or sys.stdin
-    outfile = options.outfile or sys.stdout
     sort_keys = options.sort_keys
     json_lines = options.json_lines
-    with infile, outfile:
+    with infile:
         try:
             if json_lines:
-                objs = (json.loads(line) for line in infile)
+                objs = tuple(json.loads(line) for line in infile)
             else:
                 objs = (json.load(infile), )
-            for obj in objs:
-                json.dump(obj, outfile, sort_keys=sort_keys, indent=4)
-                outfile.write('\n')
         except ValueError as e:
             raise SystemExit(e)
 
+    try:
+        outfile = sys.stdout if options.outfile == '-' else open(options.outfile, 'w')
```

Comment:
This test can be replaced by using `default=sys.stdout` and you can check something like this:
```python
from io import TextIOWrapper
....
outfile = options.outfile if isinstance(options.outfile, TextIOWrapper) else open(options.outfile, 'w')
```

Reply:
Wouldn't it be more logic to do:
```
outfile = options.outfile if options.outfile is sys.stdout else open(options.outfile, 'w')
```
?

Reply:
Yeah also works

Reply:
I used defaults but kept the types so they would be coherent with the argparse argument type. Does this seems good to you?